geode-core/src/main/java/org/apache/geode/internal/process/LocalProcessLauncher.java

```diff
@@ -96,33 +117,55 @@ void close() {
    * @throws IOException if unable to create or write to the file
    */
   private void writePid(final boolean force) throws FileAlreadyExistsException, IOException {
-    final boolean created = this.pidFile.createNewFile();
-    if (!created && !force) {
-      int otherPid = 0;
-      try {
-        otherPid = ProcessUtils.readPid(this.pidFile);
-      } catch (IOException e) {
-        // suppress
-      } catch (NumberFormatException e) {
-        // suppress
-      }
-      boolean ignorePidFile = false;
-      if (otherPid != 0 && !ignoreIsPidAlive()) {
-        ignorePidFile = !ProcessUtils.isProcessAlive(otherPid);
-      }
-      if (!ignorePidFile) {
-        throw new FileAlreadyExistsException("Pid file already exists: " + this.pidFile + " for "
-            + (otherPid > 0 ? "process " + otherPid : "unknown process"));
+    if (this.pidFile.exists()) {
+      if (!force) {
+        checkOtherPid(readOtherPid());
       }
+      this.pidFile.delete();
+    }
+
+    assert !this.pidFile.exists();
```

Comment:
I saw that you used Apache common's `Validate` class in other places.  Would that be good to use here since the default java `assert` won't do anything under normal circumstances?